pkg/yurtctl/constants/constants.go

```diff
@@ -251,5 +251,57 @@ spec:
         - name: KUBELET_SVC
           value: {{.kubeadm_conf_path}}
           {{end}}
+`
+	// DisableNodeControllerJobTemplate defines the node-controller disable job in yaml format
+	DisableNodeControllerJobTemplate = `
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: {{.jobName}}
+  namespace: kube-system
+spec:
+  template:
+    spec:
+      hostPID: true
+      hostNetwork: true
+      restartPolicy: OnFailure
+      nodeName: {{.nodeName}}
+      containers:
+      - name: yurtctl-disable-node-controller
+        image: {{.yurtctl_servant_image}}
+        imagePullPolicy: IfNotPresent
+        command:
+        - /bin/sh
+        - -c
+        args:
+        - "nsenter -t 1 -m -u -n -i -- sed -i 's/--controllers=/--controllers=-nodelifecycle,/g' {{.pod_manifest_path}}/kube-controller-manager.yaml"
+        securityContext:
+          privileged: true
+`
+	// EnableNodeControllerJobTemplate defines the node-controller enable job in yaml format
+	EnableNodeControllerJobTemplate = `
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: {{.jobName}}
+  namespace: kube-system
+spec:
+  template:
+    spec:
+      hostPID: true
+      hostNetwork: true
+      restartPolicy: OnFailure
+      nodeName: {{.nodeName}}
+      containers:
+      - name: yurtctl-enable-node-controller
+        image: {{.yurtctl_servant_image}}
+        imagePullPolicy: IfNotPresent
+        command:
+        - /bin/sh
+        - -c
+        args:
+        - "nsenter -t 1 -m -u -n -i -- sed -i 's/--controllers=-nodelifecycle,/--controllers=/g' {{.pod_manifest_path}}/kube-controller-manager.yaml"
```

Comment:
if no `--controllers` parameter is set in `kube-controller-manager.yaml`, the above command can not disable `nodelifecycle` controller.

Reply:
Oh, yes. In the cluster installed by kubeadm, the static file of kube-controller-manager contains this parameter `--controllers` by default. In order to deal with other situations, the situation without this parameter should indeed be considered.
And the method of modifying files will be optimized in the future to make string matching more accurate instead of using `sed`.